include/sys/pci.h

```diff
@@ -101,11 +104,23 @@ static inline uint32_t pci_read_config(device_t *device, unsigned reg,
   return PCI_DRIVER(device)->pci_bus.read_config(device, reg, size);
 }
 
+#define pci_read_config_1(d, r) pci_read_config((d), (r), 1)
+#define pci_read_config_2(d, r) pci_read_config((d), (r), 2)
+#define pci_read_config_4(d, r) pci_read_config((d), (r), 4)
+
 static inline void pci_write_config(device_t *device, unsigned reg,
                                     unsigned size, uint32_t value) {
   PCI_DRIVER(device)->pci_bus.write_config(device, reg, size, value);
 }
 
+#define pci_write_config_1(d, r, v) pci_write_config((d), (r), 1, v)
```

Comment:
Why `v` is not in parens, whereas `d` and `r` are?

Reply:
I misunderstood operator precedence. In fact I think we can remove parentheses around the first argument.